Tests/NooraTests/Components/SingleChoicePromptTests.swift

```diff
@@ -118,4 +118,47 @@ struct SingleChoicePromptTests {
         ✔︎ How would you like to integrate Tuist?: option1 
         """)
     }
+
+    @Test func renders_the_right_content_when_more_options_than_terminal_height() throws {
+        // Given
+        let subject = SingleChoicePrompt(
+            title: nil,
+            question: "How would you like to integrate Tuist?",
+            description: nil,
+            theme: Theme.test(),
+            terminal: terminal,
+            collapseOnSelection: true,
+            renderer: renderer,
+            standardPipelines: StandardPipelines(),
+            keyStrokeListener: keyStrokeListener
+        )
+        keyStrokeListener.keyPressStub = .init(repeating: .downArrowKey, count: 10)
+
+        // When
+        _ = subject.run(options: (1 ... 20).map { "Option \($0)" })
+
+        // Then
+        #expect(renderer.renders[0] == """
+        How would you like to integrate Tuist?
+          ❯ Option 1
+            Option 2
+            Option 3
+            Option 4
+            Option 5
+            Option 6
+            Option 7
+        ↑/↓/k/j up/down • enter confirm
+        """)
+        #expect(renderer.renders[renderer.renders.count - 2] == """
+        How would you like to integrate Tuist?
+            Option 8
+            Option 9
+            Option 10
+          ❯ Option 11
+            Option 12
+            Option 13
+            Option 14
+        ↑/↓/k/j up/down • enter confirm
+        """)
```

Comment:
nit: I'd also add a test for the render of a render when the last option is selected.